patches/minecraft/net/minecraft/world/entity/Entity.java.patch

```diff
@@ -371,6 +415,24 @@
 +      return this.m_6380_(pose, size);
 +   }
 +
++   protected FluidState touchingFluid = net.minecraft.world.level.material.Fluids.f_76191_.m_76145_();
++   protected Object2DoubleMap<FluidState> customFluidHeight = new it.unimi.dsi.fastutil.objects.Object2DoubleOpenHashMap<>();
++   public void setInFluid(FluidState touchingFluid) {
++       this.touchingFluid = touchingFluid;
++   }
++   public FluidState getTouchingFluid() {
++       return this.touchingFluid;
++   }
++   public boolean isInFluid() {
++       return this.getTouchingFluid() != net.minecraft.world.level.material.Fluids.f_76191_.m_76145_() && this.getFluidHeight(this.getTouchingFluid()) > 0.0D;
++   }
```

Comment:
I believe this method can be removed from here as it is already being added to `IForgeEntity` with the same body.